src/stackwalk.c

```diff
@@ -711,26 +715,36 @@ static void jl_print_debugloc(jl_debuginfo_t *debuginfo, jl_value_t *func, size_
     if (edges_idx) {
         jl_debuginfo_t *edge = (jl_debuginfo_t*)jl_svecref(debuginfo->edges, edges_idx - 1);
         assert(jl_typetagis(edge, jl_debuginfo_type));
-        jl_print_debugloc(edge, NULL, stmt.pc, 1);
+        jl_print_debugloc(pre_str, edge, NULL, stmt.pc, 1);
     }
     intptr_t ip2 = stmt.line;
     if (ip2 >= 0 && ip > 0 && (jl_value_t*)debuginfo->linetable != jl_nothing) {
-        jl_print_debugloc(debuginfo->linetable, func, ip2, 0);
+        jl_print_debugloc(pre_str, debuginfo->linetable, func, ip2, 0);
     }
     else {
         if (ip2 < 0) // set broken debug info to ignored
             ip2 = 0;
         const char *func_name = jl_debuginfo_name(func);
         const char *file = jl_debuginfo_file(debuginfo);
-        jl_safe_print_codeloc(func_name, file, ip2, inlined);
+        jl_safe_print_codeloc(pre_str, func_name, file, ip2, inlined);
     }
 }
 
 // Print code location for backtrace buffer entry at *bt_entry
-void jl_print_bt_entry_codeloc(jl_bt_element_t *bt_entry) JL_NOTSAFEPOINT
+void jl_print_bt_entry_codeloc(int sig, jl_bt_element_t *bt_entry) JL_NOTSAFEPOINT
 {
+    char sig_str[32], pre_str[64];
+    sig_str[0] = pre_str[0] = '\0';
+    if (sig != -1) {
+        snprintf(sig_str, 32, "signal (%d) ", sig);
+    }
+    // do not call jl_threadid if there's no current task
+    if (jl_get_current_task()) {
+        snprintf(pre_str, 64, "%sthread (%d) ", sig_str, jl_threadid() + 1);
+    }
```

Comment:
```suggestion
    int ctid = -1;
    sig_str[0] = pre_str[0] = '\0';
    if (sig != -1) {
        snprintf(sig_str, 32, "signal (%d) ", sig);
    }
    // do not call jl_threadid if there's no current task
    if (jl_get_current_task()) {
        ctid = jl_threadid() + 1;
    }
    snprintf(pre_str, 64, "%sthread (%d) ", sig_str, ctid);
```

Reply:
Without this, we don't write anything to `pre_str` if there's no current task, not even the `signal (X) ` prefix. Also, my thinking was to have the `thread (Y) ` prefix for consistency even when Y is -1.

I'm not sure if there are still cases of threads without a current task. If you're confident, you can drop the logic but we should backport the phantom task PR and test first.